mitodl/mit-open
# [Nextjs] nextjs errors

### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/5479

### Description (What does it do?)
This pr implements error catching and permission checking on the next js app

### How can this be tested?
Go to http://open.odl.local:8062/c/unit/not-a-unit. You should see a 404 Not Found Error 

As a not logged in user:
http://open.odl.local:8062/dashboard
http://open.odl.local:8062/dashboard/my-lists
http://open.odl.local:8062/dashboard/my-lists/id
http://open.odl.local:8062/dashboard/profile
http://open.odl.local:8062/dashboard/settings

Should redirect you to the ssl page

As a logged in admin
http://open.odl.local:8062/dashboard/my-lists/80000000 should show a 404 Not Found Error 

As a logged in non-admin
http://open.odl.local:8062/dashboard/my-lists/
http://open.odl.local:8062/dashboard/my-lists/1
Should display "Not Allowed"
